pkg/module/file/file.go

```diff
@@ -0,0 +1,240 @@
+	Permission string
+	// Operation pre-configures the operation type ("upload" or "download").
+	Operation string
+	// Source is the source path on either the client (for upload) or dutagent (for download).
+	Source string
+	// Destination is the destination path on either the client (for download) or dutagent (for upload).
+	Destination string
+
+	sourcePath string // sourcePath is the source file path
+	destPath   string // destPath is the destination file path
+}
+
+// Ensure implementing the Module interface.
+var _ module.Module = &File{}
+
+const abstract = `Transfer files between client and dutagent.
+`
+
+const usage = `
+ARGUMENTS:
+	<path>              Use source/destination from config, or preserve relative path structure
+	<source>:<dest>     Explicitly specify both source and destination (incompatible with source/destination config)
+
+`
+
+const description = `
+The file module transfers files between client and dutagent. The operation type
+(upload or download) must be configured in the device YAML.
+
+USAGE FORMS:
+
+1. With source/destination config:
+   - If source is configured: arg becomes the destination
+   - If destination is configured: arg becomes the source
+   - If both are configured: no arg should be provided
+   - Cannot use colon syntax when source or destination is configured
+
+2. Single path (no config):
+   - Relative paths: preserves directory structure (e.g., ./dir/file.bin → dir/file.bin)
+   - Absolute paths: uses basename only (e.g., /path/to/file.bin → file.bin)
+   Example: dutctl device cmd ./firmware.bin
+
+3. Colon syntax (explicitly specify both paths):
+   - Cannot be used if source or destination is configured
+   Example: dutctl device cmd ./firmware.bin:/custom/path.bin
+
+`
+
+func (f *File) Help() string {
+	log.Println("file module: Help called")
+
+	help := strings.Builder{}
+	help.WriteString(abstract)
+	help.WriteString(usage)
+	help.WriteString(fmt.Sprintf("Configured operation is %q.\n", f.Operation))
+
+	if f.Source != "" {
+		help.WriteString(fmt.Sprintf("Configured source path is %q.\n", f.Source))
+	}
+
+	if f.Destination != "" {
+		help.WriteString(fmt.Sprintf("Configured destination path is %q.\n", f.Destination))
+	}
+
+	if f.Permission != "" {
+		help.WriteString(fmt.Sprintf("Configured file permission is %q.\n", f.Permission))
+	} else {
+		help.WriteString("Default file permission is \"0644\".\n")
+	}
+
+	help.WriteString(description)
+
+	return help.String()
+}
+
+func (f *File) Init() error {
+	log.Println("file module: Init called")
+
+	// Validate configuration
+	return f.validateConfig()
+}
+
+func (f *File) Deinit() error {
+	log.Println("file module: Deinit called")
+
+	return nil
+}
+
+func (f *File) Run(_ context.Context, sesh module.Session, args ...string) error {
+	log.Println("file module: Run called")
+
+	path, err := f.evalArgs(args)
+	if err != nil {
+		return err
+	}
+
+	// Parse paths
+	err = f.parsePaths(path)
+	if err != nil {
+		return err
+	}
```

Comment:
Would prefer one function* taking `arg` and having source and destination set afterward. 
*_may it be a function or a method to `f`_